File: 2107-find-unique-binary-string/find-unique-binary-string.py
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        n = len(nums)
        answer = ''
        def backtracking(now, len_now):
            if len_now == n:
                return '' if now in nums else now

            for i in ['0', '1']:
                answer = backtracking(now+i, len_now+1)
                if answer != '':
                    return answer
            
            return answer
        
        return backtracking('',0)

File: 2107-find-unique-binary-string/find-unique-binary-string.java
import java.util.*;

class Solution {
    public String findDifferentBinaryString(String[] nums) {
        int n = nums.length;

        int[] arr = new int[n];

        for(int i=0; i<n; i++) {
            arr[i] = Integer.parseInt(nums[i], 2);
        }
        Arrays.sort(arr);

        int maxN = (int) Math.pow(2, n);

        int i = 0;
        int j = 0;
        while(j <= maxN && i < n) {
            System.out.println(arr[i] + " " + j);

            if(arr[i] == j) {
                i++;
                j++;
            } 
            else {

                return String.format("%0"+n+"d", Integer.parseInt(Integer.toBinaryString(j)));
            }
        }
        return String.format("%0"+n+"d", Integer.parseInt(Integer.toBinaryString(j)));
    }
}